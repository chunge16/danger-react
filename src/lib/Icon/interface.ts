import {SVGAttributes, CSSProperties} from 'react';

export interface IconProps extends Omit<SVGAttributes<SVGElement>, 'className'> {
  style?: CSSProperties;
  /**
   * 图标类型
   */
  type: string;
  /**
   * 是否旋转
   */
  spin?: boolean;
  // /**
  //  * 节点类名
  //  */
  className?: string | string[];
}


export interface CustomIconComponentProps {
  style?: CSSProperties;
  className?: string;
  width?: string | number;
  height?: string | number;
  fill?: string;
  viewBox?: string;
}
