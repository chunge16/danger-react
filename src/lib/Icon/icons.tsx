import React from 'react';
import Icon from './index';
import dangerIconData from '@/assets/icons/danger-icon.json';
import './index.css';

function Icons(){
  let {glyphs} = dangerIconData;
  if (!glyphs) return <div>数据错误</div>

  return (
      <div
          className='Icons-warp'
      >
        {
          glyphs.map(item => {
            return (
                <div
                    className="Icons-item"
                    key={item.name}
                >
                  <Icon type={`danger-icon-${item.name}`} />
                  <label htmlFor="id" style={{ marginTop: '8px' }}>
                    {item.name}
                  </label>
                </div>
            )
          })
        }
      </div>
  )
}


export default Icons
