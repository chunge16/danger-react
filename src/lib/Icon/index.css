 @keyframes spinAnimation {
     to {
         transform: rotate(360deg);
     }
 }

 .spin-animation {
     animation: spinAnimation 1s linear infinite;
 }


 .Icons-warp {
     display: flex;
     flex-wrap: wrap;
     & .Icons-item {
         margin-right: 8px;
         margin-bottom: 8px;
         padding: 16px;
         width: 7.5rem;
         display: flex;
         flex-direction: column;
         align-items: center;
         background-color: #fff;
         color: #000;
         border-radius: 4px;
         cursor: pointer;
         &:hover {
             box-shadow: #e8e8e8 0 0 2px 2px;
         }
     }
 }
