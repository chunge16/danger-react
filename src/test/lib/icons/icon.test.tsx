import {beforeEach, describe, expect, test} from 'vitest';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';

import Icon from '@/lib/icons';

describe("Icon test", () => {

  test('Icon should show iconauto', () => {
    render(<Icon data-testid={'test-icon'} name={'iconauto'}/>);

    let iconDom = screen.queryByTestId('test-icon');
    let className = iconDom?.getAttribute('class');
    expect(iconDom).toBeDefined();
    expect(className).toEqual('svg-icon')
  })

  // class 存在 icon-spin-animation 说明 spin 动画成功
  test('Icon test props spin', async () => {
    const {container} = await render(<Icon spin data-testid={'test-icon'} name={'iconauto'}/>);
    const actionDom = container.querySelector('.action');

    expect(actionDom).not.toBeEmptyDOMElement();
    expect(actionDom).toBeInTheDocument();
    expect(actionDom?.classList?.contains('icon-spin-animation')).toEqual(true)

  })

})
