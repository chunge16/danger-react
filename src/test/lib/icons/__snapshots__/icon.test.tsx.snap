// Vitest Snapshot v1

exports[`Icon test > Icon should render 1`] = `
<div>
  <span
    class="action"
  >
    <svg
      class="svg-icon"
      data-testid="test-icon"
    >
      <use
        xlink:href="#iconauto"
      />
    </svg>
  </span>
</div>
`;
